### DISCUSSION TITLE
rke2-coredns-rke2-coredns became rke2-coredns

### DISCUSSION THREAD

**Discussion by albundy83** [rancher/rke2#5491] [Q&A] [answered]
**Environmental Info:**
RKE2 Version: 

v1.27.11-rc1+rke2r1

Node(s) CPU architecture, OS, and Version: 

Linux ip-10-0-1-20 6.5.0-18-generic #18~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Feb  7 11:40:03 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:

3 control-planes, 2 workers

**Describe the bug:**

I'm not sure if it's a bug but when I move to release v1.27.11-rc1+rke2r1 (I used to be on v1.26.12+rke2r1), rke2-coredns-rke2-coredns Service name became rke2-coredns.
ConfigMap, Pods also became rke2-coredns instead of rke2-coredns-rke2-coredns.
rke2-coredns-rke2-coredns-autoscaler also rename to rke2-coredns-autoscaler.
Well, you got the idea :)

Maybe also it's the update of the rke2-coredns component as we move from 1.24.008 to 1.29.000

Sorry, I did not find more informations to determine what change this name.

**Steps To Reproduce:**

- Installed RKE2: 

**Expected behavior:**

Nothing special, I just want to be sure that it's something expected

**Actual behavior:**

**Additional context / logs:**

**Comment by brandond** [comment]
> when I move to release v1.27.11-rc1+rke2r1 (I used to be on v1.26.12+rke2r1), rke2-coredns-rke2-coredns Service name became rke2-coredns.
ConfigMap, Pods also became rke2-coredns instead of rke2-coredns-rke2-coredns.
rke2-coredns-rke2-coredns-autoscaler also rename to rke2-coredns-autoscaler.

Is there a problem, or are you just asking if this is expected? I suspect that this was a change to the upstream chart; we didn't intentionally alter that but it sounds reasonable.

Ref: https://github.com/rancher/rke2-charts/commit/56596f9958f70296dec30f17c6d95bfc1e2690f0

**Reply by albundy83** [reply]
Well, as I told you, I don't know what to expect of this modification.
I'm not sure about the impact neither as I did not try it yet on a production cluster.
I discover it as my loki helm chart has this parameter set:
```yaml
global:
  dnsService: rke2-coredns-rke2-coredns
```
and of course I add to change it to:
```yaml
global:
  dnsService: rke2-coredns
```

because of the error:
```
 host not found in resolver "rke2-coredns-rke2-coredns.kube-system.svc.cluster.local." in /etc/nginx/nginx.conf:33
```
Yeah, I spend 1 hour understanding why my loki-gateway did not work anymore :)

My remark was more to be sure that this update was intentional and to notify users about it.

**Reply by brandond** [reply, marked as answer]
Looks like this came from https://github.com/coredns/helm/pull/156

This can cause problems when upgrading, due to helm not sequencing the deletion of the old service and creation of the new service properly, leading to clusterIP conflicts (https://github.com/rancher/rke2/issues/5492). We'll be reverting this in our repackaging of the chart.

https://github.com/rancher/rke2-charts/pull/414

**Reply by albundy83** [reply]
Sound good to me, thanks a lot :)
